### PR TITLE
NatsCustodian technical information

### PR DESCRIPTION
**Pull request by LaurensVergote**

### CODE REVIEW

**Comment by LaurensVergote** [comment]
@MykytaDeyneha

File: user-guide/Reference/Technical_information/NatsCustodian.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    S11[Detect incorrect routes] --> Q1
+    Q1{Detected changes for <br/> automatic Nats reconfiguration?} -->|Yes| A1
+    Q1 -->|No| Q2
+    A1[Automatic Nats reconfiguration] --> A1S1
+    A1S1[Reconfigure Nats] --> A1S2
+    A1S2[Restart Nats] --> Q2
+    Q2{Detected changes for<br/>manual Nats reconfiguration?} -->|Yes| A2
+    Q2 -->|No| Q3
+    A2[Manual Nats reconfiguration] --> Q3
+    Q3{Detected changes for Nats restart?} -->|No| End
```
[the remark this answers is not in the document]
Comment (LaurensVergote) [reply user-guide/Reference/Technical_information/NatsCustodian.md]: indeed

[the remark this answers is not in the document]
Comment (LaurensVergote) [reply user-guide/Reference/Technical_information/NatsCustodian.md]: Done

File: user-guide/Reference/Technical_information/NatsCustodian.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    Q3 -->|Yes| A3
+    A3[Restart Nats] --> End
+```
+
+## Manual Nats Reconfiguration
+
+TODO: Document the BPA
```
[the remark this answers is not in the document]
Comment (LaurensVergote) [reply user-guide/Reference/Technical_information/NatsCustodian.md]: Done

**Review by lustpieter — APPROVED** [review]

**Review by mykebueno — COMMENTED** [review]
Approved but what about the ## NatsCustodian workflow: Starting from DataMiner 10.2.0-CU18 / 10.3.0-CU6 / 10.3.9 part?

File: user-guide/Reference/Technical_information/NatsCustodian.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+The system incorporates several checks, including the detection of new nodes, tracking of deleted nodes, identification of unavailable nodes, verification of configuration accuracy, validation of NATS process operations, and detection of incorrect routes.
+
+If any inconsistencies are identified during these checks, a NATS reconfiguration or restart is triggered. A Nats reconfiguration can be either manual or automatic, depending on the type of anomaly detected. The appearance of new nodes or removal of existing ones triggers an automatic reconfiguration. 
+
+Other detected inconsistencies are logged by a BPA. These are usually non-critical and do not trigger an automatic reconfiguration.
+
+This approach is designed to give administrators the opportunity to explore and understand the root cause of issues before undertaking a reconfiguration, fostering a more educated approach to system adjustments and mitigating potential disruptions to active operations. The detection of incorrect routes will trigger a Nats restart to reestablish the correct routes to the local Nats server.
```
[the remark this answers is not in the document]
Comment (LaurensVergote) [reply user-guide/Reference/Technical_information/NatsCustodian.md:43]: > Should we leave the text regarding the BPA? Does it belong here?

I suppose we can merge as is and document the BPA when it is merged?

[the remark this answers is not in the document]
Comment (mykebueno) [reply user-guide/Reference/Technical_information/NatsCustodian.md:43]: ops just correcting this last line: **an automatic NATS reconfiguration is initiated**

[the remark this answers is not in the document]
Comment (LaurensVergote) [reply user-guide/Reference/Technical_information/NatsCustodian.md]: Done

**Review by mykebueno — APPROVED** [review]

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**